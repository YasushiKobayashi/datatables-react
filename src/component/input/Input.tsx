import * as React from 'react';
import { css } from 'emotion';

interface Props {
  label?: string;
  value: string;
  handleInput: (v: string) => void;
}

const id = 'react-datatable-input';

const styleWrapper = css({
  display: 'flex',
  '& input': {
    width: '100%',
  },
});

const Input: React.FC<Props> = props => {
  const { label, value, handleInput } = props;

  const labelContent = label ? <label htmlFor={id}>{label}</label> : null;

  return (
    <div className={styleWrapper}>
      {labelContent}
      <input
        type="text"
        id={id}
        data-testid={id}
        className={id}
        onChange={e => handleInput(e.currentTarget.value)}
        value={value}
      />
    </div>
  );
};

export { Input as default };
