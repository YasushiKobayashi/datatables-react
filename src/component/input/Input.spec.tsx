import * as React from 'react';
import * as Enzyme from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';

import Input from './Input';

Enzyme.configure({ adapter: new Adapter() });

const value = 'value';

const base = {
  value,
};

const id = 'react-datatable-input';

describe('Input', () => {
  it('handleInput', () => {
    const handleInput = jest.fn();
    const props = Object.assign({}, base, { handleInput });
    const content = Enzyme.mount(<Input {...props} />);

    const target = content.find(`input[data-test="${id}"]`);
    expect(target.get(0).props.value).toBe(value);

    target.simulate('change', { target: { value: id } });
    expect(handleInput).toBeCalled();
  });

  it('renders correctly', () => {
    const handleInput = jest.fn();
    const props = Object.assign({}, base, { handleInput });
    const tree = Enzyme.shallow(<Input {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
