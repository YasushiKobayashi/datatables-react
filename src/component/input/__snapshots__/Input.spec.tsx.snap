// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders correctly 1`] = `
<DocumentFragment>
  <div
    class="css-1b7vfsy"
  >
    <input
      class="react-datatable-input"
      data-testid="react-datatable-input"
      id="react-datatable-input"
      type="text"
      value="value"
    />
  </div>
</DocumentFragment>
`;
